aea/cli/interact.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Implementation of the 'aea interact' subcommand."""
+
+import codecs
+from typing import Optional
+
+import click
+
+from aea.configurations.base import PublicId
+from aea.connections.stub.connection import (
+    DEFAULT_INPUT_FILE_NAME,
+    DEFAULT_OUTPUT_FILE_NAME,
+    StubConnection,
+)
+from aea.mail.base import Envelope, InBox, Multiplexer, OutBox
+
+
+class InterruptInputException(Exception):
+    """An exception to mark an interuption event."""
+
+
+def try_construct_envelope() -> Optional[Envelope]:
```

Comment:
First iteration of a command to interact with a running aea via the stub connection. Helps during development...